### ISSUE TITLE
Some Buttons have small edge regions where they animate, but don't work, and vice-versa

### ISSUE THREAD

**Issue by jhey29** [ppy/osu#28232]
### Type

Cosmetic

### Bug description

When I use the Mod Menu and the editor, I move my cursor towards a button I want to click until that button does its hover animation, then click. I expect the button to then work.
If the button is the Test Button in the Editor, or the Back, Mod Customization, Deselect All (probably), or + (add Personal Preset) Button in the Mod Menu, it has a small region to the right and to the bottom of it where it animates if my cursor is there, but clicking doesn't do anything _except play the click animation._ This is annoying for the Test Button because that region coincides with the bottom edge of the screen, which it is very easy to put your cursor at. As a bonus, at least the Mod Menu Back Button also has a region on it's top and left where hovering and clicking plays no animations, but closes the menu anyways.
Possibly unrelated details are that I'm on Ubuntu and using the "High Precision Mouse" feature and a laptop touchpad to move my cursor (this might make the problem slightly more noticeable because clicking by tapping does not make the cursor move at all). There are other buttons without this problem though.
This problem has been there for a little while, so it's nothing new, which makes sense considering that these buttons are older than my 3-month osu! career. 

### Screenshots or videos

![osu_2024-05-19_15-02-27](https://github.com/ppy/osu/assets/110559474/60a01733-3436-4f48-83c3-fb82629e2929)
![osu_2024-05-19_15-04-35](https://github.com/ppy/osu/assets/110559474/dbb104ca-1151-4357-af69-9df1c8f2bc51)
![osu_2024-05-19_15-05-23](https://github.com/ppy/osu/assets/110559474/ac115d5c-3480-4013-8c0f-37f4216402cc)
![osu_2024-05-19_15-08-10](https://github.com/ppy/osu/assets/110559474/2e62764f-5835-4bd5-99d8-4726822900f3)

### Version

2024.519.1

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/15369502/compressed-logs.zip)

**Comment by bdach** [comment]
Pretty sure this is caused by the scale change on mouse down:

https://github.com/ppy/osu/blob/dedc2d2aa0bf40045a5abd8830e3402433bb44ac/osu.Game/Graphics/UserInterface/ShearedButton.cs#L175-L179

Due to the trainwreck that is `OsuClickableContainer`:

https://github.com/ppy/osu/blob/dedc2d2aa0bf40045a5abd8830e3402433bb44ac/osu.Game/Graphics/Containers/OsuClickableContainer.cs#L21-L25

this means that after the initial mouse down the button shrinks such that the mouse position is now outside content, and as such the button no longer receives positional input, and thus never gets a click triggered for it.

The solution here is probably to nuke `OsuClickableContainer`. I'm not sure what real gains it provides at this point. It can be reimplemented in around 10 lines of code.

**Comment by jhey29** [comment]
you might have noticed that the top/left region I identified is actually just intended behaviour in the Mod Menu: there are a bunch of places

**Comment by bdach** [comment]
> you might have noticed that the top/left region I identified is actually just intended behaviour in the Mod Menu: there are a bunch of places like e.g. the entire footer, where clicking closes the mod menu

Yeah clicking "the background" is supposed to exit out of the screen. The footer itself isn't supposed to do this but the gap between it and the columns is.

**Comment by jhey29** [comment]
yeah, correct. the actual issue with the scaling and the edges also happens for the beat snap divisor switching and play buttons in the editor-
